enhancements/network/user-defined-network-segmentation.md

```diff
@@ -413,6 +521,202 @@ guarantee uniqueness and eliminates the ability to falsify access.
 After creating the CRD, check the status to ensure OVN-Kubernetes has accepted this network to serve the namespaces
 selected. Now tenants may go ahead and be provisioned their namespace.
 
+#### Controller
+
+New controller should be introduced, it should  manage the new CRDs lifecycle.
+
+The controller should utilize the CNI plugin API and create OVN networks using `NetworkAttachmentDefinition` using the 
+`ovn-k8s-overlay-cni` CNI plugin according to desired spec.
+
+The controller should watch NAD object in order to reconcile them and reflect the network state in the CRDs status.
+
+The request corresponding NADs should be created with a finalizer, so that upon deletion OVN-Kubernetes can validate that 
+there are no pods still using this network.
+If there are pods still attached to this network, the network will not be removed.
+
+The CRD spec should be validated before creating the NAD, see the [validation](#validations) section for more details.
+
+The controller should create the requested NAD with:
+1. finalizer, enable the controller release resources before NAD is deleted, ensure no pod is connected to the network.
+2. owner-reference referring to the request CR object.
+   Using the owner-reference mechanism should prevent deletion of the NAD before the corresponding CRD instance is deleted.
+   In addition, the owner-reference make teardown seamless, when the request CR object instance is deleted,
+   the cluster garbage collected will dispose the corresponding NAD when all finalizers are gone.
+
+In a scenario a NAD already exist at the target namespace, the controller should check if the existing NAD is managed 
+by the contoller as follows:
+1. Check the owner reference match the request CR UID.
+2. OVN-K user-defined-network finalizer exist.
+3. NAD spec correspond to desired spec.
+In case one of the following checks fails, the controller should reconcile the request and 
+this state in the CR status network is not ready.
+
+In scenario of cluster-scope CRD request, the controller should do best effort to create the desired NAD in each
+specified namespace.
+In case on or more NAD creation fails (e.g.: due to namespace not exist), the controller should continue to the 
+next namespace.
+When finished, reflect in the status all namespaces where NAD creation failed to create.
+
+##### General flow
+1. On namespaced-scope CRD creation:
+  1.1. Validates the CRD spec. 
+  1.2. Generate NAD manifest from the CRD spec.
+  1.3. Check no the desired NAD is not created yet, if so, create the NAD and return.
+  1.4. Otherwise, verify the existing NAD correspond to desired spec, if so return.
+  1.5. In case the NAD is malformed raise an error and return.
+  1.6. Update the status as follows:
+   1.6.1. Reflect namespaces where network is available.
```

Comment:
well on namespace scoped CRD there can only be one namespace where the network is available, so this status should be more simple than the cluster scoped.